examples/custom_search_method/asha_search_method/core_search_runner/searcher.yaml

```diff
@@ -0,0 +1,14 @@
+name: core-search-runner
+entrypoint: python3 core_search_runner/run_experiment.py
+hyperparameters:
+  max_length: 1000
+  max_trials: 16
+  num_rungs: 3
+  divisor: 4
+searcher:
+  metric: validation_error
+  smaller_is_better: true
+  name: single
+  max_length:
+    batches: 100
+max_restarts: 0
```

Comment:
github complaining about missing end of line

Reply:
what do you think about moving hyperparameter.max_trials to be the searcher.max_length so in experiment code searcher.max_length is not ignored but passed to ASHA as the max_trials?

Reply:
That's a good idea in theory :) but in practice it requires accessing searcher config through `_config`, e.g., `det.ExperimentConfig(info.trial._config)` or `get_cluster_info().trial._config['searcher']['metric']` which according to our code
```
 _config is private because it's not a stable API; as the experiment config version increases, the shape of the experiment config dict might change as the master is upgraded. As a result, we should strongly discourage users from reading the experiment config dict in their training code.
```
Long story short, I don't think we should advertise `_config`. I don't know a different (safe) way of accessing searcher metric, but if there is one, I can update the example.